### PR TITLE
Added Cycle sort

### PR DESCRIPTION
**Pull request by pavitraag**
## Related Issue
Added cycle sort

## Type of PR

- [ ] Bug fix
- [X] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [] I have commented my code, particularly in hard-to-understand areas.
- [X] I have followed the code style guidelines of this project.
- [X] I have checked for any existing open issues that my pull request may address.
- [X] I have ensured that my changes do not break any existing functionality.
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X] I have read the resources for guidance listed below.
- [X] I have followed security best practices in my code changes.

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

**Comment by pavitraag** [comment]
Closes #2845

File: dsa-solutions/CycleSort.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  - Beginner
+  - Sorting Algorithms
+  - Geeks for Geeks
+  - CPP
+  - Python
+  - Java
+  - JavaScript
+  - DSA
+description: "This is a solution to the Bubble Sort problem on Geeks for Geeks."
+---
```
Comment (Ajay-Dhangar) [inline dsa-solutions/CycleSort.md:15]: Please add some content to this file or delete it if it's not needed. We do not require empty files.

File: dsa-solutions/Sorting-Algorithms/Cycle-Sort.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+-Third cycle: Place 50 at index 4 (correct position).[10, 20, 30, 40, 50]
+-Fourth and fifth cycles: Remaining elements are already in the correct position.
+-Final sorted array: [10, 20, 30, 40, 50].
+
+## 6. key constraints and characteristics of Cycle Sort:
+
+- **Memory Writes**: Minimizes the number of write operations, making it efficient where write operations are costly.
+- **Stability**: Maintains the relative order of equal elements, making it a stable sorting algorithm.
+- **Performance**:
+  - Time Complexity: O(n^2) in the worst case.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Cycle-Sort.md:75]: replace `O(n^2)` to `$O(n^2)$`

File: dsa-solutions/Sorting-Algorithms/Cycle-Sort.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  - Space Complexity: O(1) additional space, as it sorts in-place.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Cycle-Sort.md:76]: replace `O(1)` to `$O(1)$`

File: dsa-solutions/Sorting-Algorithms/Cycle-Sort.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  
+- **Stable Sorting:** Preserves the order of equal elements, ensuring stability in sorting results.
+
+- **In-Place Algorithm:** Sorts elements directly within the input array, requiring minimal extra space.
+
+- **Suitable for Small Data Sets:** Efficient for small datasets or nearly sorted arrays due to direct placement of elements.
+
+### Disadvantages:
+
+- **Quadratic Time Complexity:** Has O(n^2) time complexity in the worst-case scenario, limiting efficiency for large datasets.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/Sorting-Algorithms/Cycle-Sort.md:197]: replace `O(n^2)` to `$O(n^2)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by pavitraag** [comment]
@Ajay-Dhangar Resolved the issues

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**